src/components/puzzle/Tile.js

```diff
@@ -1,25 +1,67 @@
-import React, { PureComponent } from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import ClassNames from 'classnames';
+import Media from 'react-media';
+
+import { durationTranslate } from '../../config';
 
 import Fade from '../transitions/Fade';
 
-export default class Tile extends PureComponent {
+import { dirFromMove } from '../../core/game';
+
+const defaultTransitionStyle = {
+    transition: `transform ${durationTranslate}ms ease-out`,
+};
+
+const MAX_TILE_SIZE = 100;
+
+const maxTileSizeStyle = nbTiles => ({
+    backgroundSize: `${(nbTiles + 1) * MAX_TILE_SIZE}px`,
+    height: `${MAX_TILE_SIZE}px`,
+    width: `${MAX_TILE_SIZE}px`,
+});
+
+const buildTranslateStyle = (
+    grid,
+    tileMargin,
+    tileSize,
+    tileValue,
+    unit = 'vw',
+) => {
+    const dir = dirFromMove(grid, tileValue);
+    const translatingDir = {
```

Comment:
Extract all computation based on the grid size to a new file full of pure JS functions